spec/models/spree/payment/processing_decorator_spec.rb

```diff
@@ -14,5 +14,12 @@
 
       expect(payment.close!).to be_truthy
     end
+
+    it 'payment receives close! method' do
+      payment = create(:amazon_payment, state: 'completed')
+
+      expect(payment).to receive(:close!)
```

Comment:
Yeah, you need to do it the other way around - stub it before it comes to it:

```ruby
expect(payment.source).to receive(:close!)
payment.close!
```

and that's it